src/envs/behavior_options.py

```diff
@@ -0,0 +1,1012 @@
+                            euler_angles = np.array([0.0, np.pi, 0.0])
+
+                    state = p.saveState()
+                    # plan a motion to the pose [x, y, z, euler_angles[0],
+                    # euler_angles[1], euler_angles[2]]
+                    plan = plan_hand_motion_br(
+                        robot=env.robots[0],
+                        obj_in_hand=None,
+                        end_conf=[
+                            x,
+                            y,
+                            z,
+                            euler_angles[0],
+                            euler_angles[1],
+                            euler_angles[2],
+                        ],
+                        hand_limits=((minx, miny, minz), (maxx, maxy, maxz)),
+                        obstacles=get_body_ids(env,
+                                               include_self=True,
+                                               grasping_with_right=True),
+                        rng=rng,
+                    )
+                    p.restoreState(state)
+
+                    # NOTE: This below line is *VERY* important after the
+                    # pybullet state is restored. The hands keep an internal
+                    # track of their state, and if we don't reset their this
+                    # state to mirror the actual pybullet state, the hand will
+                    # think its elsewhere and update incorrectly accordingly
+                    env.robots[0].parts["right_hand"].set_position(
+                        env.robots[0].parts["right_hand"].get_position())
+                    env.robots[0].parts["left_hand"].set_position(
+                        env.robots[0].parts["left_hand"].get_position())
+
+                    # Grasping Phase 2: Move along the vector from the
+                    # position the hand ends up in to the object and
+                    # then try to grasp.
+
+                    if plan is not None:
+                        hand_pos = plan[-1][0:3]
+                        hand_orn = plan[-1][3:6]
+                        # Get the closest point on the object's bounding
+                        # box at which we can try to put the hand
+                        closest_point_on_aabb = get_closest_point_on_aabb(
+                            hand_pos, lo, hi)
+
+                        delta_pos_to_obj = [
+                            closest_point_on_aabb[0] - hand_pos[0],
+                            closest_point_on_aabb[1] - hand_pos[1],
+                            closest_point_on_aabb[2] - hand_pos[2],
+                        ]
+                        delta_step_to_obj = [
+                            delta_pos / 25.0 for delta_pos in delta_pos_to_obj
+                        ]  # we want to accomplish the motion in 25 timesteps
+
+                        # move the hand along the vector to the object until it
+                        # touches the object
+                        for _ in range(25):
```

Comment:
just curious, how important exactly is this 25 parameter to tune correctly? seems like it can't be either too small or too large, else bad things happen?

Reply:
not particularly; it just dictates the speed at which we move the hand. 25 was my first (and arbitrary) choice.